File: bldr/web/boot/codec.ts
import {
  BootReport,
  type BootReport as BootReportMessage,
} from './report.pb.js'

// marshalBootReportJson encodes a BootReport as deterministic proto JSON.
export function marshalBootReportJson(report: BootReportMessage): string {
  return BootReport.toJsonString(report)
}

File: bldr/web/boot/codec.go
package boot

import protojson "github.com/aperturerobotics/protobuf-go-lite/json"

// MarshalBootReportJSON encodes a BootReport as deterministic proto JSON.
func MarshalBootReportJSON(report *BootReport) ([]byte, error) {
	return (protojson.MarshalerConfig{EnumsAsInts: false}).Marshal(report)
}
